test/sys/test_socket.rs

```diff
@@ -1509,11 +1509,20 @@ fn test_scm_credentials_and_rights() {
 #[test]
 fn test_too_large_cmsgspace() {
     let space = vec![0u8; 1024];
-    test_impl_scm_credentials_and_rights(space);
+    test_impl_scm_credentials_and_rights(space).unwrap();
 }
 
 #[cfg(linux_android)]
-fn test_impl_scm_credentials_and_rights(mut space: Vec<u8>) {
+#[test]
+fn test_too_small_cmsgspace() {
+    let space = vec![0u8; 4];
+    assert!(test_impl_scm_credentials_and_rights(space).is_err());
```

Comment:
Let's assert it with `ENOBUF` if that is the expected value:

```suggestion
    assert_eq!(test_impl_scm_credentials_and_rights(space), Err(nix::errno::Errno::ENOBUF));
```